lib/bitcoin/lock/LockResolver.ts

```diff
@@ -0,0 +1,139 @@
+   */
+  public async resolveLockIdentifierAndThrowOnError (lockIdentifier: LockIdentifier): Promise<BlockchainLockModel> {
+
+    // The verifictation of a lock-identifier has the following steps:
+    //   (A). The redeem script in the lock-identifier is actually a 'locking' script
+    //   (B). The redeem script in the lock-identifier is paying to the wallet in the lock-identifier
+    //   (C). The transaction in the lock-identifier is paying to the redeem script in the lock-identifier
+    //
+    // With above, we can verify that the amount is/was locked for the specified wallet in
+    // the specified transaction.
+
+    // (A). verify redeem script is a lock script
+    const redeemScriptObj = LockResolver.createScriptFromHexInput(lockIdentifier.redeemScriptAsHex);
+    const [isLockScript, lockUntilBlock] = LockResolver.isRedeemScriptALockScript(redeemScriptObj);
+
+    if (!isLockScript) {
+      throw new BitcoinError(ErrorCode.LockResolverRedeemScriptIsNotLock, `${redeemScriptObj.toASM()}`);
+    }
+
+    // (B). verify that the script is paying to the target wallet
+    const walletAddressObj = new Address(lockIdentifier.walletAddressAsBuffer);
+    const isLockScriptPayingTargetWallet = LockResolver.isRedeemScriptPayingToTargetWallet(redeemScriptObj, walletAddressObj);
+
+    if (!isLockScriptPayingTargetWallet) {
+      throw new BitcoinError(ErrorCode.LockResolverRedeemScriptIsNotPayingToWallet,
+                            `Script: ${redeemScriptObj.toASM()}; Wallet: ${walletAddressObj.toString()}`);
+    }
+
+    // (C). verify that the transaction is paying to the target redeem script
+    const lockTransaction = await this.bitcoinClient.getRawTransaction(lockIdentifier.transactionId);
+
+    const transactionIsPayingToTargetRedeemScript = lockTransaction.outputs.length > 0 &&
+                                                    LockResolver.isOutputPayingToTargetScript(lockTransaction.outputs[0], redeemScriptObj);
+
+    if (!transactionIsPayingToTargetRedeemScript) {
+      throw new BitcoinError(ErrorCode.LockResolverTransactionIsNotPayingToScript,
+                             `Transaction id: ${lockIdentifier.transactionId} Script: ${redeemScriptObj.toASM()}`);
+    }
+
+    // Now that the lock identifier has been verified, return the lock information
+    const serializedLockIdentifier = LockIdentifierSerializer.serialize(lockIdentifier);
+
+    return {
+      identifier: serializedLockIdentifier,
+      amountLocked: lockTransaction.outputs[0].satoshis,
+      lockEndTransactionTime: lockUntilBlock,
+      linkedWalletAddress: walletAddressObj.toString()
+    };
+  }
+
+  /**
+   * Checks whether the redeem script is indeed a lock script.
+   * @param redeemScript The script to check.
+   * @returns A touple where [0] == true if the script is a lock script; false otherwise, [1] == if [0] is true then the lockUntilBlock.

```

Comment:
```suggestion
   * @returns A touple where [0] = true and [1] = lock-until-block if the script is a lock script; [0] = false and [1] = 0 otherwise.
```